torch/nn/_functions/vision.py

```diff
@@ -2,11 +2,12 @@
 import torch.backends.cudnn as cudnn
 
 
-def affine_grid_generator(theta, size):
-    # type: (Tensor, List[int]) -> Tensor
-    if theta.is_cuda and cudnn.enabled and cudnn.is_acceptable(theta) and len(size) == 4 and size[0] < 65536:
+def affine_grid_generator(theta, size, align_corners):
+    # type: (Tensor, List[int], bool) -> Tensor
+    if (theta.is_cuda and cudnn.enabled and cudnn.is_acceptable(theta)
+            and len(size) == 4 and size[0] < 65536 and align_corners):
         N, C, H, W = size
         ret = torch.cudnn_affine_grid_generator(theta, N, C, H, W)
```

Comment:
I agree that the dispatch to `cudnn` should be removed if it doesn't add performance gain.
I would also, by the way, advocate for completely getting rid of this file `torch/nn/_functions/vision.py` and having the `affine_grid()` function of `nn/functional.py` call directly into the C++ backend.

But I think both of these changes should happen in a separate pull request, since they are not related to this pull request's core purpose, and might require a separate discussion.